flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriter.java

```diff
@@ -46,8 +50,8 @@ private FlinkOrcWriter(RowType rowType, Schema iSchema) {
     }
   }
 
-  public static OrcRowWriter<RowData> buildWriter(RowType rowType, Schema iSchema) {
-    return new FlinkOrcWriter(rowType, iSchema);
+  public static OrcRowWriter<RowData> buildWriter(RowType rowType, Schema iSchema, TypeDescription schema) {
```

Comment:
is it necesSary to pass in the type description? can we get id from the iceberg schema?

Reply:
Yeah I tried to avoid changing the signature but wasn't able to find id information from schema, I'll see if people more familiar with the project would have comment on this

Reply:
The field IDs are kept in struct fields, not primitives. That's probably why you didn't find one that was usable in the `primitive` method. What we do in other visitors is add methods to the visitor that are called before and after visiting a struct field, array element, map key, and map value. Those methods are passed the field. Then the visitor just needs to implement the before/after to maintain a stack of field IDs.

Here's a `visit` method with the callbacks: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/TypeUtil.java#L334
And here's an example of using them to get the field IDs: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java#L75-L83

Reply:
Thanks for the info! Yeah earlier I noticed that the ids exist in `NestedField ` but wasn't able to find out a good way to extract without larger changes to the signature, and tried to replace `StructType`/`NestedType` with `NestedField` but that would result in losing other information. I'll update to use the before/after pattern.